R/graded.R

```diff
@@ -252,8 +252,23 @@ fail <- function(
   hint = getOption("gradethis.fail.hint", FALSE),
   encourage = getOption("gradethis.fail.encourage", FALSE)
 ) {
+  user_provided_hint <- !missing(hint)
+  user_provided_message <- !missing(message)
```

Comment:
Should we use `rlang::missing_arg()` instead?

And have the "arg is missing" value be the respective `getOption()` call above

Reply:
I considered that (and I think we'd also be fine with a sentinel `NULL` here), but I think it somewhat obscures the default value or how the default value is determined.

I think I'd rather add a helper for looking up global default values, e.g. `gradethis_default()`

```r
fail <- function(
  message = gradethis_default("fail"),
  ...,
  hint = gradethis_default("fail.hint"),
  encourage = gradethis_default("fail.encourage")
) { ... }
```

where the documentation can better direct users to `gradethis_setup()` (they'd be in the same help page) where those values are set.

Internally, `gradethis_default()` could add an `attr()` that we could use in cases like this while still keeping the default values in the function signature.

Reply:
Btw, I'm envisioning the above as a separate chunk of work 😄

Reply:
Given that is on a list somewhere, sounds good.

Reply:
https://github.com/rstudio/gradethis/issues/245